projects/bootloader/test/test_bootloader_crc32.c

```diff
@@ -0,0 +1,55 @@
+#include <stdlib.h>
+
+#include "bootloader_crc32.h"
+#include "bootloader_mcu.h"
+#include "config.h"
+#include "crc32.h"
+#include "flash.h"
+#include "interrupt.h"
+#include "log.h"
+#include "persist.h"
+#include "test_helpers.h"
+#include "unity.h"
+
+uint32_t s_crc32_codes_size = 54;
+
+// intialize memory with random numbers for testing
+void intialize_memory() {
+  size_t curr_size = 0;
+  // writing the memory using a buffer with 2048 bytes at a time
+  uint8_t buffer[2048];
+  unsigned int seed = 42;
+
+  // generating random values in range of uint8(0->255)
+  for (int i = 0; i < 2048; i++) buffer[i] = rand_r(&seed) % (255 + 1 - 0) + 0;
```

Comment:
I'm pretty sure `rand_r` is a Linux thing (so not available on stm32), CI is choking over it.  Use `rand` from <stdlib> instead, and seed it with `srand`.

Reply:
done